### PR TITLE
[exporter/datadog] Move configuration structs to main package

### PR DESCRIPTION
**Pull request by mx-psi**
**Description:** 

- Move configuration structs to main package of exporter. Deprecate `config` package.
- Remove deprecated `Sanitize` method by making it private (renamed to `logWarnings`).

**Link to tracking Issue:** Updates #8373

### CODE REVIEW

File: exporter/datadogexporter/config.go
Comment (mx-psi) [inline exporter/datadogexporter/config.go:451]: This file was copied over from the `config` package. The only change is on this function (done on the second commit): `Sanitize` was renamed to `logWarnings`, making it unexported.

**Review by TylerHelmuth — APPROVED** [review]

**Review by codeboten — COMMENTED** [review]
Please rebase

File: exporter/datadogexporter/factory.go
Comment (gbbr) [inline exporter/datadogexporter/factory.go:177]: Do we need these empty lines at the beginning of both `create*Exporter` funcs?

File: exporter/datadogexporter/factory.go
Comment (gbbr) [inline exporter/datadogexporter/factory.go:238]: Same q here.

File: exporter/datadogexporter/config.go
Comment (gbbr) [inline exporter/datadogexporter/config.go:494]: What is this method? What is its purpose? Why is it here? Can we document it?

File: exporter/datadogexporter/config.go
Comment (gbbr) [reply exporter/datadogexporter/config.go:451]: Can we not log those in `Validate()`?

**Review by gbbr — COMMENTED** [review]
This is a good change, I think 🙂

**Review by gbbr — APPROVED** [review]

File: exporter/datadogexporter/factory.go
Comment (mx-psi) [reply exporter/datadogexporter/factory.go]: Removed

File: exporter/datadogexporter/factory.go
Comment (mx-psi) [reply exporter/datadogexporter/factory.go]: Removed

File: exporter/datadogexporter/factory.go
Comment (mx-psi) [reply exporter/datadogexporter/factory.go]: That's a good idea, done in 02fe02053125c8af7c13a41a5be350d27f6bc9db

File: exporter/datadogexporter/config.go
Comment (mx-psi) [reply exporter/datadogexporter/config.go:500]: It implements this interface: https://pkg.go.dev/go.opentelemetry.io/collector/config#Unmarshallable I added a comment and an interface assertion

**Comment by mx-psi** [comment]
@codeboten did you want to leave a review here?

**Comment by TylerHelmuth** [comment]
> @djaglowski This PR needs two release notes: one for deprecating the `config` package, and another for the removal of the `Sanitize` method. The two are closely related and are tracked by the same issue. What should I do? 

Related to this comment: https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/11847#discussion_r911473399. For now I think you have to modify the changelog manually and then add a skip changelog label. Then then the release manager had to do some manual merging during the release.

**Comment by mx-psi** [comment]
Thanks @TylerHelmuth! I added `Skip Changelog` to bypass the changelog check, since that also sounds like a valid solution. I will wait for Daniel to see this before merging just in case :)

**Comment by djaglowski** [comment]
I agree w/ what Tyler said for now. Longer term, this is obviously not ideal.

Should we just remove the requirement that only 1 new entry is added? I think it's worth continuing to reject modifications/removals, but adding multiple change entries is probably allowable. I'm not sure we necessarily need to optimize for this case either, since it is infrequent enough. It seems like adding two files with some duplication between them is not unreasonable. WDYT?

**Comment by mx-psi** [comment]
> Should we just remove the requirement that only 1 new entry is added? I think it's worth continuing to reject modifications/removals, but adding multiple change entries is probably allowable. I'm not sure we necessarily need to optimize for this case either, since it is infrequent enough. It seems like adding two files with some duplication between them is not unreasonable. WDYT?

I think that makes sense, yes, thanks for having a look! Modifications/removals are more likely to be mistakes (although there could be legitimate uses but they are very rare)

**Pull request merged**
